pkg/controller/messaginguser/messaginguser_controller.go

```diff
@@ -0,0 +1,354 @@
+	"github.com/enmasseproject/enmasse/pkg/keycloak"
+	"github.com/enmasseproject/enmasse/pkg/util"
+	"github.com/enmasseproject/enmasse/pkg/util/finalizer"
+	"github.com/enmasseproject/enmasse/pkg/util/recon"
+
+	logr "github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_messaginguser")
+var _ reconcile.Reconciler = &ReconcileMessagingUser{}
+
+const (
+	ANNOTATION_REALM_NAME = "enmasse.io/realm-name"
+	FINALIZER_NAME        = "enmasse.io/standard-authservice"
+)
+
+type ReconcileMessagingUser struct {
+	client    client.Client
+	reader    client.Reader
+	scheme    *runtime.Scheme
+	namespace string
+
+	newKeycloakClientFunc keycloak.NewKeycloakClientFunc
+	keycloakClients       map[string]keycloak.KeycloakClient
+}
+
+// Gets called by parent "init", adding as to the manager
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+func newReconciler(mgr manager.Manager) *ReconcileMessagingUser {
+	return &ReconcileMessagingUser{
+		client:                mgr.GetClient(),
+		reader:                mgr.GetAPIReader(),
+		scheme:                mgr.GetScheme(),
+		newKeycloakClientFunc: keycloak.NewClient,
+		keycloakClients:       make(map[string]keycloak.KeycloakClient),
+		namespace:             util.GetEnvOrDefault("NAMESPACE", "enmasse-infra"),
+	}
+}
+
+func add(mgr manager.Manager, r *ReconcileMessagingUser) error {
+
+	// Create a new controller
+	c, err := controller.New("messaginguser-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource MessagingUser
+	err = c.Watch(&source.Kind{Type: &userv1beta1.MessagingUser{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileMessagingUser) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling MessagingUser")
+
+	ctx := context.TODO()
+
+	user := &userv1beta1.MessagingUser{}
+	err := r.client.Get(ctx, request.NamespacedName, user)
+	if err != nil {
+		if k8errors.IsNotFound(err) {
+			reqLogger.Info("MessagingUser resource not found. Ignoring since object must be deleted")
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Failed to get MessagingUser")
+		return reconcile.Result{}, err
+	}
+
+	rc := &recon.ReconcileContext{}
+
+	reqLogger.V(8).Info("checkFinalizer", "user", user)
```

Comment:
We are logging the user object that may includes the password.  This could end of being log scraped and thus cause security issues.  Is there a way we can simply avoid printing the password filled (Stringer perhaps)? or maybe just log the object metadata?